mobilecoind/src/sync.rs

```diff
@@ -603,6 +615,193 @@ mod test {
         assert!(!utxos.contains(&first_utxo));
     }
 
+    // TODO: make this a bench instead of a unit test.
+    #[test_with_logger]
+    fn test_sync_monitor_with_random_recipients(logger: Logger) {
+        let mut rng: StdRng = SeedableRng::from_seed([98u8; 32]);
+
+        let account_keys: Vec<_> = (0..5).map(|_i| AccountKey::random(&mut rng)).collect();
+
+        let data = MonitorData::new(
+            account_keys[0].clone(),
+            DEFAULT_SUBADDRESS_INDEX, // first subaddress
+            5,                        // number of subaddresses
+            0,                        // first block
+            "",                       // name
+        )
+        .unwrap();
```

Comment:
```suggestion
        let first_subaddress = DEFAULT_SUBADDRESS_INDEX;
        let subaddress_count = 5;
        let first_block_index = 0;
        let name = "";
        let data = MonitorData::new(
            account_keys[0].clone(),
            first_subaddress,
            subaddress_count,
            first_block_index,
            name,
        )
        .unwrap();
```
I think it's usually better to use descriptive variable names rather than comments because the latter become stale more easily.